### ISSUE TITLE
What is the idea behind making the the `owner` private in Ownable.sol?

### ISSUE THREAD

**Issue by frozeman** [OpenZeppelin/openzeppelin-contracts#2293]
What is the idea behind making the the `owner` private in https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/access/Ownable.sol#L19?

In most cases the owner should be set by the overwriting contract, rather than the `msg.sender` as in the internal constructor.

What is the thinking behind this choice?

**Comment by frozeman** [comment]
Ok I figured it out. I can simply call:
```js
constructor(address _newOwner) public {
        transferOwnership(_newOwner);

        ...
    }
```
In a constructor of the derived contract.

Though this will be slightly more costly as also the event is fired twice (once from 0x0 to msg.sender and then to the owner given in the derived constructor). I saw this as less ideal, definitely makes the depleting cost a bit more expensive than it would need to be.
